src/main/java/org/snomed/aag/data/services/CriteriaItemService.java

```diff
@@ -175,17 +175,30 @@ public void verifyManual(CriteriaItem criteriaItem, boolean expectedManual) {
 	}
 
 	/**
-	 * Remove entry from collection. If entry does not have a corresponding flag in Branch metadata
-	 * or the corresponding flag in Branch metadata is false, then remove entry from collection.
+	 * If entry does not have a corresponding flag in Branch metadata or the corresponding flag in
+	 * Branch metadata is false, then remove entry from collection. If the corresponding mandatory flag
+	 * is present and true, the entry will be marked as mandatory.
 	 *
 	 * @param criteriaItems Collection to process
 	 * @param branch        Branch to cross reference
 	 */
-	public void removeNonEnabled(Set<CriteriaItem> criteriaItems, Branch branch) {
+	public void removeNonEnabledMarkMandatory(Set<CriteriaItem> criteriaItems, Branch branch) {
 		verifyParams(criteriaItems, branch);
 		Set<String> authorFlags = MetadataUtil.getEnabledAuthorFlags(branch);
 		if (!authorFlags.isEmpty()) {
-			criteriaItems.removeIf(item -> item.getEnabledByFlag().stream().noneMatch(authorFlags::contains));
+			criteriaItems.removeIf(item -> {
+				Set<String> enabledByFlag = item.getEnabledByFlag();
+				boolean correspondingFlag = enabledByFlag.stream().anyMatch(authorFlags::contains);
+				if (!correspondingFlag) {
+					return true;
+				}
+
+				if (enabledByFlag.contains(Constants.MANDATORY_FLAG)) {
```

Comment:
I don't think we need this bit.
The mandatory flag can be used for item selection. The enabledByFlag field can be used to enable the item. I think they are independant. I don't think we have a use case for a enabledByFlag called "mandatory" - sorry if I have confused you!